src/twisted/names/client.py

```diff
@@ -174,7 +174,10 @@ def parseConfig(self, resolvConf):
         for L in resolvConf:
             L = L.strip()
             if L.startswith(b'nameserver'):
-                resolver = (nativeString(L.split()[1]), dns.PORT)
+                parsedString = L.split()[1]
+                if b'.' in parsedString:
+                    parsedString = b'::' + parsedString
```

Comment:
So it looks like if you see a line like:

nameserver 128.0.0.1

you are changing the address to

::128.0.0.1

This seems wrong also, and would explain why some of the tests are failing.